helm: Add priorityClass with default system-node-critical to spire co…

Tilusch opened the pull request:
Currently the Helmchart to install Cilium does not allow to set a priorityClassName to the spire agent or server. 
Since those are mission critical component when using cilium service mesh it would be a good practice to set a corresponding priorityClassName.


```release-note
Add default prioriyClass system-node-critical to spire components
```

gandro commented on install/kubernetes/cilium/templates/_helpers.tpl:
```diff
@@ -58,6 +58,27 @@ where:
 {{- end -}}
 
+{{/*
+Return user specify priorityClass or default criticalPriorityClass
```
Is there any reason we can't use `cilium.priorityClass` instead of defining a new one? The helper seems identitical?

Tilusch replied:
I just added the new helper because it looked better for me. But indeed those helpers are identical, i will remove the spire helper again and use the cilium helper instead.

Tilusch replied:
@gandro Now the cilium helper gets used again. Hope this looks better now :)

gandro replied:
Great, thank you!

gandro reviewed (commented):
Thanks! Helm parts look good, just one question

Tilusch committed:
helm: Add priorityClass to spire components

Addet default priorityClassName system-node-critical to the spiro components and added the capability to overwrite the spiro components priorityClassName

Signed-off-by: Tilusch <til.heini@swisscom.com>
